lib/grape-swagger/doc_methods/optional_object.rb

```diff
@@ -4,11 +4,23 @@ class OptionalObject
       class << self
         def build(key, options, request = nil)
           if options[key]
-            options[key].is_a?(Proc) ? options[key].call : options[key]
+            return evaluate(key, options, request) if options[key].is_a?(Proc)
+            options[key]
           else
-            request
+            request.send(default_values[key])
           end
         end
+
+        def evaluate(key, options, request)
+          options[key].arity == 0 ? options[key].call : options[key].call(request)
```

Comment:
I am not in love with this. IMO this should be an API that takes a fixed number of parameters. Maybe we should also make the parameter a hash right away so that we can pass more than just the request in here?